ibm/service/catalogmanagement/resource_ibm_cm_offering_instance.go

```diff
@@ -1,150 +1,225 @@
-// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Copyright IBM Corp. 2022 All Rights Reserved.
 // Licensed under the Mozilla Public License v2.0
 
 package catalogmanagement
 
 import (
+	"context"
 	"fmt"
 	"log"
-	"os"
-	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 
 	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
 	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM/go-sdk-core/v5/core"
 	"github.com/IBM/platform-services-go-sdk/catalogmanagementv1"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-)
-
-const (
-	inProgress = "in progress"
-	failed     = "failed"
-	success    = "succeeded"
-
-	waitUntilInterval = 10 * time.Second
 )
 
 func ResourceIBMCmOfferingInstance() *schema.Resource {
 	return &schema.Resource{
-		Create:   resourceIBMCmOfferingInstanceCreate,
-		Read:     resourceIBMCmOfferingInstanceRead,
-		Update:   resourceIBMCmOfferingInstanceUpdate,
-		Delete:   resourceIBMCmOfferingInstanceDelete,
-		Exists:   resourceIBMCmOfferingInstanceExists,
-		Importer: &schema.ResourceImporter{},
-
-		Timeouts: &schema.ResourceTimeout{
-			Create: schema.DefaultTimeout(4 * time.Minute),
-			Update: schema.DefaultTimeout(4 * time.Minute),
-			Delete: schema.DefaultTimeout(4 * time.Minute),
-		},
+		CreateContext: resourceIBMCmOfferingInstanceCreate,
+		ReadContext:   resourceIBMCmOfferingInstanceRead,
+		UpdateContext: resourceIBMCmOfferingInstanceUpdate,
+		DeleteContext: resourceIBMCmOfferingInstanceDelete,
+		Importer:      &schema.ResourceImporter{},
 
 		Schema: map[string]*schema.Schema{
-			"url": {
+			"x_auth_refresh_token": &schema.Schema{
 				Type:        schema.TypeString,
-				Computed:    true,
+				Required:    true,
+				Description: "IAM Refresh token.",
+			},
+			"rev": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Cloudant revision.",
+			},
+			"url": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
 				Description: "url reference to this object.",
 			},
-			"crn": {
+			"crn": &schema.Schema{
 				Type:        schema.TypeString,
-				Computed:    true,
+				Optional:    true,
 				Description: "platform CRN for this instance.",
 			},
-			"label": {
+			"label": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "the label for this instance.",
 			},
-			"catalog_id": {
+			"catalog_id": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
```

Comment:
why is the catalog_id which us required made as optional. Can we execute this resource without this argument?
Many arguments which are required are maked as optional